diff --git a/navit/graphics/gtk_drawing_area/graphics_gtk_drawing_area.c b/navit/graphics/gtk_drawing_area/graphics_gtk_drawing_area.c
index 6ec67d1..8e0ef30 100644
--- a/navit/graphics/gtk_drawing_area/graphics_gtk_drawing_area.c
+++ b/navit/graphics/gtk_drawing_area/graphics_gtk_drawing_area.c
@@ -609,7 +609,7 @@ configure(GtkWidget * widget, GdkEventConfigure * event, gpointer user_data)
 		cairo_destroy(gra->cairo);
 	gra->cairo = cairo_create(surface);
 	cairo_surface_destroy(surface);
-	cairo_set_antialias (gra->cairo, CAIRO_ANTIALIAS_GOOD);
+	cairo_set_antialias (gra->cairo, CAIRO_ANTIALIAS_DEFAULT);
 	callback_list_call_attr_2(gra->cbl, attr_resize, GINT_TO_POINTER(gra->width), GINT_TO_POINTER(gra->height));
 	return TRUE;
 }
